src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -468,7 +468,7 @@ private static Sampler GetSampler(IConfiguration configuration, Sampler? stateSa
     private static double ReadTraceIdRatio(IConfiguration configuration)
     {
         if (configuration.TryGetStringValue(TracesSamplerArgConfigKey, out var configValue) &&
-                double.TryParse(configValue, out var traceIdRatio))
+                double.TryParse(configValue, System.Globalization.NumberStyles.Float | System.Globalization.NumberStyles.AllowThousands, System.Globalization.CultureInfo.InvariantCulture, out var traceIdRatio))
```

Comment:
No need of `AllowThousands` for parsing traceIdRatio.

```suggestion
                double.TryParse(configValue, System.Globalization.NumberStyles.Float, System.Globalization.CultureInfo.InvariantCulture, out var traceIdRatio))
```